tests/networkpolicy_test.go

```diff
@@ -0,0 +1,246 @@
+	"fmt"
+
+	"github.com/google/goexpect"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1network "k8s.io/api/networking/v1"
+	v13 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("Networkpolicy", func() {
+
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	var vmia *v1.VirtualMachineInstance
+	var vmib *v1.VirtualMachineInstance
+	var vmic *v1.VirtualMachineInstance
+
+	tests.BeforeAll(func() {
+		tests.SkipIfUseFlannel(virtClient)
+		tests.BeforeTestCleanup()
+		// Create three vmis, vmia and vmib are in same namespace, vmic is in different namespace
+		vmia = tests.NewRandomVMIWithEphemeralDiskAndUserdataAndNS(tests.RegistryDiskFor(tests.RegistryDiskCirros), tests.NamespaceTestDefault, "#!/bin/bash\necho 'hello'\n")
+		vmia.Labels = map[string]string{"type": "test"}
+		vmia, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmia)
+		Expect(err).ToNot(HaveOccurred())
+		tests.WaitForSuccessfulVMIStart(vmia)
+
+		vmib = tests.NewRandomVMIWithEphemeralDiskAndUserdataAndNS(tests.RegistryDiskFor(tests.RegistryDiskCirros), tests.NamespaceTestDefault, "#!/bin/bash\necho 'hello'\n")
+		_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmib)
+		Expect(err).ToNot(HaveOccurred())
+		tests.WaitForSuccessfulVMIStart(vmib)
+
+		vmic = tests.NewRandomVMIWithEphemeralDiskAndUserdataAndNS(tests.RegistryDiskFor(tests.RegistryDiskCirros), tests.NamespaceTestAlternative, "#!/bin/bash\necho 'hello'\n")
+		_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestAlternative).Create(vmic)
+		Expect(err).ToNot(HaveOccurred())
+		tests.WaitForSuccessfulVMIStart(vmic)
+
+		vmia, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmia.Name, &v13.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		vmib, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmib.Name, &v13.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		vmic, err = virtClient.VirtualMachineInstance(tests.NamespaceTestAlternative).Get(vmic.Name, &v13.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+	})
+
+	Context("vms limited by Default-deny networkpolicy", func() {
+
+		BeforeEach(func() {
+			// deny-by-default networkpolicy will deny all the traffice to the vms in the namespace
+			By("Create deny-by-default networkpolicy")
+			networkpolicy := &v1network.NetworkPolicy{
+				ObjectMeta: v13.ObjectMeta{
+					Name: "deny-by-default",
+				},
+				Spec: v1network.NetworkPolicySpec{
+					PodSelector: v13.LabelSelector{},
+					Ingress:     []v1network.NetworkPolicyIngressRule{},
+				},
+			}
+			_, err := virtClient.NetworkingV1().NetworkPolicies(vmia.Namespace).Create(networkpolicy)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(" should be failed to reach vmia from vmib", func() {
+			By("Connect vmia from vmib")
```

Comment:
I see a lot of duplication, can you create the nested method under `Describe` that will
- log to some VMI
- send some cmd input
- check cmd output

When VMI, input, and output can be method arguments.

Reply:
I have updated the code according your comments. Please review again. Thanks:)